easybuild/easyblocks/s/scalapack.py

```diff
@@ -104,11 +105,19 @@ def build_libscalapack_make(self):
         extra_makeopts = []
 
         acml = get_software_root(Acml.LAPACK_MODULE_NAME[0])
+        flexiblas = get_software_root(FlexiBLAS.LAPACK_MODULE_NAME[0])
+        intelmkl = get_software_root(IntelMKL.LAPACK_MODULE_NAME[0])
         lapack = get_software_root(Lapack.LAPACK_MODULE_NAME[0])
         openblas = get_software_root(OpenBLAS.LAPACK_MODULE_NAME[0])
-        intelmkl = get_software_root(IntelMKL.LAPACK_MODULE_NAME[0])
 
-        if lapack:
+        if flexiblas:
```

Comment:
I agree with this but it in general I'm wondering about what we should do with flexiblas in this scenario since you are guaranteed to get positive results from `get_software_root` for OpenBLAS as well as FlexiBLAS? This means that order matters in how things are decided...but how do we enforce this? It also means that if I give FlexiBLAS in the dependencies when it is not in the toolchain, then it get's preferred, whereas if I give OpenBLAS in the deps it does not get preferred.

Perhaps we need a helper function to grab a single library in a generic way?

Reply:
This decision is going to be required for the cmakemake easyblock, so that might be the right place to implement it.

Reply:
For ScaLAPACK specifically, there is no BLAS/LAPACK in the toolchain, that's provided as a dependency.

I check for FlexiBLAS first here exactly to avoid that OpenBLAS kicks in first.

We can come up with theoretical scenarios where this could be a limitation, but I'm not sure those will actually pop up.

One way around it could be to add a custom easyconfig parameter here so you can explicitly specify to which library you want ScaLAPACK to be linked, and only "guess" (like we do now) if that's not specified. But is that worth implementing here?

Reply:
Here, I think things are fine, but we will need something (reusable) for CMakeMake